MAGE-1201 fix recommend model validation

cammonro opened the pull request:
**Summary**

This PR:
- Adds friendlier error messaging for 404 errors including "no index found"
- Removes dependency on `renderingContent` - not a good measure of valid model
- Adds soft fail warning on missing hits on configured model
- Fixes validation on Looking Similar 

TODO:
- Address store scoped model validation 

**Result**

<img width="1383" alt="image" src="https://github.com/user-attachments/assets/dc2d0a60-e1c4-4caf-95c4-157ee9de48ee" />

damcou commented on Observer/RecommendSettings.php:
```diff
@@ -154,12 +159,46 @@ protected function validateRecommendation(string $changedPath, string $recommend
+    protected function shouldDisplayWarning(array $recommendationResponse): bool
+    {
+        return
+            $this->appState->getAreaCode() === \Magento\Framework\App\Area::AREA_ADMINHTML
```
Why do we need to check this ? Any reasons why the area code could be different ?

cammonro replied:
The event observed by this class is dispatched by `Magento\Config\Model\Config::save` so it's possible it might not be triggered from the admin panel (perhaps CLI scripts, web API's, other customizations etc.). I wanted to play it safe so as not to introduce any unwanted side effects.

damcou replied:
ok makes sense :)

damcou commented on Model/RecommendManagement.php:
```diff
@@ -75,7 +75,7 @@ public function getTrendingItemsRecommendation(): array
     public function getLookingSimilarRecommendation(string $productId): array
     {
-        return $this->getRecommendations($productId, 'bought-together');
+        return $this->getRecommendations($productId, 'looking-similar');
```
🙄

cammonro replied:
LOL

damcou commented on Observer/RecommendSettings.php:
```diff
@@ -143,9 +147,10 @@ protected function validateRecommendation(string $changedPath, string $recommend
-                throw new LocalizedException(__(
-                    "It appears that there is no trained model available for Algolia application ID %1.",
+            if ($this->shouldDisplayWarning($recommendations)) {
+                $this->messageManager->addWarningMessage(__(
```
It's way better like this, I guess the models can take a while to be trained so displaying a warning instead of an error makes definitely more sense.

damcou reviewed (approved):
LGTM ✅ 
As discussed on JIRA, this is still not ideal and needs additional love, but at least this should unblock some users.

cammonro commented:
@damcou Thanks for the feedback here! Until we can implement proper store scoping on this validation I've gone and removed the hard fail across the board and switched to issuing warnings only.
